packages/jaeger-ui/src/reducers/trace.test.js

```diff
@@ -18,56 +18,194 @@ import traceGenerator from '../demo/trace-generators';
+      payload: { data: [trace] },
+      meta: { id },
+    });
+    expect(state.traces).toEqual({ [id]: { id, data: transformTraceData(trace), state: fetchedState.DONE } });
+  });
+
+  it('handles a failed FETCH_TRACE', () => {
+    const error = new Error();
+    const state = traceReducer(undefined, {
+      type: `${jaegerApiActions.fetchTrace}${ACTION_POSTFIX_REJECTED}`,
+      payload: error,
+      meta: { id },
+    });
+    expect(state.traces).toEqual({ [id]: { error, id, state: fetchedState.ERROR } });
+    expect(state.traces[id].error).toBe(error);
   });
-  expect(state.traces).toEqual({ [id]: { id, data: transformTraceData(trace), state: fetchedState.DONE } });
 });
 
-it('trace reducer should handle a failed FETCH_TRACE', () => {
-  const error = new Error();
-  const state = traceReducer(undefined, {
-    type: `${jaegerApiActions.fetchTrace}_REJECTED`,
-    payload: error,
-    meta: { id },
+describe('fetch multiple traces', () => {
+  const traceB = traceGenerator.trace({ numberOfSpans: 1 });
+  const { traceID: idB } = traceB;
+
+  it('sets loading to true for all pending IDs', () => {
+    const traces = { preExisting: Math.random() };
+    const state = traceReducer(
+      { traces },
+      {
+        type: `${jaegerApiActions.fetchMultipleTraces}${ACTION_POSTFIX_PENDING}`,
+        meta: { ids: [id, idB] },
+      }
+    );
+    const outcome = {
+      ...traces,
+      [id]: { id, state: fetchedState.LOADING },
+      [idB]: { id: idB, state: fetchedState.LOADING },
+    };
+    expect(state.traces).toEqual(outcome);
+  });
+
+  describe('handles a successful request', () => {
+    it('transforms and saves all trace data', () => {
+      const traces = { preExisting: Math.random() };
+      const state = traceReducer(
+        { traces },
+        {
+          type: `${jaegerApiActions.fetchMultipleTraces}${ACTION_POSTFIX_FULFILLED}`,
+          payload: { data: [trace, traceB] },
+        }
+      );
+      const outcome = {
+        ...traces,
+        [id]: { id, data: transformTraceData(trace), state: fetchedState.DONE },
+        [idB]: { id: idB, data: transformTraceData(traceB), state: fetchedState.DONE },
+      };
+      expect(state.traces).toEqual(outcome);
+    });
+
+    it('saves all error data', () => {
+      const msg = 'a-message';
+      const traceID = 'a-trace-id';
+      const traces = { preExisting: Math.random() };
+      const state = traceReducer(
+        { traces },
+        {
+          type: `${jaegerApiActions.fetchMultipleTraces}${ACTION_POSTFIX_FULFILLED}`,
+          payload: { data: [], errors: [{ msg, traceID }] },
+        }
+      );
+      const outcome = {
+        ...traces,
+        [traceID]: { id: traceID, error: expect.any(Error), state: fetchedState.ERROR },
+      };
+      expect(state.traces).toEqual(outcome);
+    });
+  });
+
+  it('handles a failed request', () => {
+    const error = 'error-info';
+    const traces = { preExisting: Math.random() };
```

Comment:
here and other places:
why use `Math.random()` as "filler data"? It seems as though it'd be clearer to simply write `{ preExistingID: 'preExistingTraceDate' }`